akka-stream/src/main/scala/akka/stream/Shape.scala

```diff
@@ -26,13 +26,26 @@ sealed abstract class OutPort
  * is the InPort (which does not bear an element type because Modules only
  * express the internal structural hierarchy of stream topologies).
  */
-final class Inlet[-T](override val toString: String) extends InPort
+object Inlet {
+  def apply[T](toString: String): Inlet[T] = new Inlet[T](toString)
+}
+
+final class Inlet[-T] private (override val toString: String) extends InPort {
+  def carbonCopy[R](): Inlet[R] = Inlet(toString)
```

Comment:
True, the rationale was merely to be able to switch the type of an Inlet (given that they have no behavior).
I'll switch to remove the type parameter.